src/trufont/objects/factories.py

```diff
@@ -96,7 +96,36 @@ def pathGraphicsPathFactory(path):
             stack.append(point.y)
             if point.type == "curve":
                 graphicsPath.AddCurveToPoint(*stack)
-                stack = []
+                stack.clear()
+            #  If we reach here, there are (assuming a valid contour) three
+            #  possible states:
+            #
+            #  1. We have one off-curve on the stack and we are looking at a
+            #     qcurve: simply draw the curve with the control point being the
+            #     off-curve on the stack.
+            #  2. We have more than one off-curve on the stack and are looking at
+            #     a qcurve. Here, TrueType's implied on-curve principle applies
+            #     that requires that an implied on-curve point is inserted
+            #     halfway between all off-curve points. The last off-curve is
+            #     then used as the control-point of the qcurve.
+            #  3. The stack is full of off-curves and we have reached the last
+            #     point. An all off-curve path is to be handled like a quadratic
+            #     path.
+            elif point.type == "qcurve" or point is points[-1]:
+                if len(stack) == 1:
+                    graphicsPath.AddQuadCurveToPoint(*stack)
+                    stack.clear()
+                else:
+                    stack_offcurves = trufont.util.misc.grouper(stack, 2)
+                    for (cx1, cy1), (cx2, cy2) in trufont.util.misc.pairwise(
+                        stack_offcurves
+                    ):
```

Comment:
- This assumes number of points is even, right?
- That's a lot of intermediary functions for a simple iteration, we can instead do something like:

```python
offset = 0
size = len(stack)
while 4 + offset < size:
  points = stack[offset:4+offset]
  ...
  offset += 2
```

Can be rewritten in terms of for loop as e.g.:

```python
for offset in range(0, len(stack) - 4, 2):
  points = stack[offset:4+offset]
```